Sources/Caching/FileRepository.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  FileRepository.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/13/25.
+
+import Foundation
+
+/// A file cache
+@_spi(Internal) public class FileRepository: @unchecked Sendable {
+    let networkService: SimpleNetworkService
+
+    private let store = KeyedDeferredValueStore<InputURL, OutputURL>()
+    private let fileManager: Caching
+
+    private lazy var cacheDirectory: URL? = fileManager.cacheDirectory
```

Comment:
I think we can set this in the initializer, to avoid the laziness and make it inmutable, right?

Reply:
Also, wouldn't making it inmutable make `FileRepository: Sendable`?